libsolidity/formal/SMTChecker.cpp

```diff
@@ -724,3 +747,24 @@ smt::Expression SMTChecker::var(Declaration const& _decl)
 	solAssert(m_variables.count(&_decl), "");
 	return m_variables.at(&_decl);
 }
+
+void SMTChecker::popPathCondition()
+{
+	solAssert(m_pathConditions.size() > 0, "Cannot pop path condition, empty.");
+	m_pathConditions.pop_back();
+}
+
+void SMTChecker::pushPathCondition(smt::Expression const& _e)
+{
+	if (m_pathConditions.size() == 0)
```

Comment:
Perhaps use `m_pathConditions.push_back(getCurrentPathConditions() && _e)` so that we have the case-distinction only in one function?